src/Form/GoogleAuthSettingsForm.php

```diff
@@ -27,43 +63,88 @@ public function getFormId() {
   /**
    * {@inheritdoc}
    */
+  protected function getEditableConfigNames() {
+    return array_merge(
+      parent::getEditableConfigNames(),
+      ['social_auth_google.settings']
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function buildForm(array $form, FormStateInterface $form_state) {
     $config = $this->config('social_auth_google.settings');
 
     $form['google_settings'] = [
       '#type' => 'details',
-      '#title' => $this->t('Google Client settings'),
+      '#title' => $this->t('Google App settings'),
       '#open' => TRUE,
+      '#description' => $this->t('You need to first create a Google App at <a href="@google-dev">@google-dev</a>', ['@google-dev' => 'https://developers.google.com/apps']),
     ];
 
     $form['google_settings']['client_id'] = [
       '#type' => 'textfield',
       '#required' => TRUE,
       '#title' => $this->t('Client ID'),
       '#default_value' => $config->get('client_id'),
-      '#description' => $this->t('Copy the Client ID here'),
+      '#description' => $this->t('Copy the Client ID of your Google App here. This value can be found from your App Dashboard.'),
     ];
 
     $form['google_settings']['client_secret'] = [
       '#type' => 'textfield',
       '#required' => TRUE,
       '#title' => $this->t('Client Secret'),
       '#default_value' => $config->get('client_secret'),
-      '#description' => $this->t('Copy the Client Secret here'),
+      '#description' => $this->t('Copy the Client Secret of your Google App here. This value can be found from your App Dashboard.'),
+    ];
+
+
+    $form['google_settings']['data_points'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Data Points To Be Collected'),
+      '#default_value' => 'name,email',
+      '#description' => $this->t('Define the data point to be stored in database, data points must be separated by comma.'),
+    ];
+
+    $form['google_settings']['oauth_redirect_url'] = [
+      '#type' => 'textfield',
+      '#disabled' => TRUE,
+      '#title' => $this->t('Valid OAuth redirect URIs'),
+      '#description' => $this->t('Copy this value to <em>Valid OAuth redirect URIs</em> field of your Google App settings.'),
+      '#default_value' => $GLOBALS['base_url'] . '/user/login/google/callback',
+    ];
+
+    $form['google_settings']['app_domains'] = [
+      '#type' => 'textfield',
+      '#disabled' => TRUE,
+      '#title' => $this->t('App Domains'),
```

Comment:
Google calls it "Authorized JavaScript origins". Please use the appropriate definition for each implementer